python/lib/candidate.py

```diff
@@ -83,27 +83,66 @@ def create_candidate(self, db, participants_info):
             if 'age' in row:
                 self.age = row['age']
 
+            # three steps to find site:
+            #   1. try matching full name from 'site' column in participants.tsv in db
+            #   2. try extracting alias from pscid
+            #   3. try finding previous site in candidate table
+
             if 'site' in row:
+                # search site id in psc table by its full name
                 site_info = db.pselect(
                     "SELECT CenterID FROM psc WHERE Name = %s",
                     [row['site'], ]
                 )
                 if(len(site_info) > 0):
                     self.center_id = site_info[0]['CenterID']
-            else:
-                db_sites = db.pselect("SELECT CenterID, Alias FROM psc")
-                for site in db_sites:
-                    if site['Alias'] in row['participant_id']:
-                        self.center_id = site['CenterID']
+
+            if self.center_id is None:
+                # search site id in psc table by its alias extracted from pscid
+                p = re.compile(r'[^a-zA-Z]+', re.UNICODE)
+                possible_site = p.sub('', self.psc_id)[2:].upper()
+                db_site = db.pselect(
+                    "SELECT CenterID, Alias FROM psc WHERE Alias = %s",
+                    [possible_site, ]
+                )
+                if(len(db_site) > 0):
+                    self.center_id = db_site[0]['CenterID']
+
+            if self.center_id is None:
+                # try to find participant site in db
+                candidate_site_project = db.pselect(
+                    "SELECT RegistrationCenterID FROM candidate WHERE pscid = %s",
+                    [self.psc_id, ]
+                )
+                try:
```

Comment:
We can avoid a try/catch block by checking the length of the result: if(len(candidate_site_project) > 0):